### ISSUE TITLE
uninitialized constant ActionView::Template::Handlers::ERB::Erubi (NameError)

### ISSUE THREAD

**Issue by wyattisimo** [haml/haml#1011]
I'm getting "uninitialized constant ActionView::Template::Handlers::ERB::Erubi (NameError)" on line 37 of haml/lib/haml/railtie.rb: https://github.com/haml/haml/blob/c1b38ed87319a0c23e0247a3eb574cf88bce8bb5/lib/haml/railtie.rb#L36-L37

Dumping some context...

```ruby
const_defined?('::Erubi') #=> true

defined?(::Erubi) #=> constant

const_defined?('ActionView::Template::Handlers::ERB::Erubi') #=> true

defined?(ActionView::Template::Handlers::ERB::Erubi) #=> nil
```

I'm using MRI v2.6.3 and Rails v4.2.11.1. Any clue what's going on here? Why would `const_defined?` say a constant by that name exists, but `defined?` doesn't recognize it? Is there possibly a scoping issue?

**Comment by nickhammond** [comment]
@wyattisimo Some useful context around `Erubi` in https://github.com/haml/haml/issues/946, similar issue.

**Comment by amatsuda** [comment]
Thank you for reporting this!

This seems like a ruby's bug to me. https://bugs.ruby-lang.org/issues/15875
But for now we need a fix in Haml's side not to hit this `const_defined?` bug.

**Comment by amatsuda** [comment]
So, this used to be happening only when you're using rails 4.2.x with haml 5.1.0, and bundling erubi or something that depends on erubi (e.g. better_rrors), and running them on ruby 2.6.1-2.6.3.

Now it's been fixed via 9374849ce50ee68692d0fd21f088d1ab19b45bd0 and 09cb8efa1ce4e242b0ef1e05ce40fc6fc7b5d4c1, hopefully.

**Comment by amatsuda** [comment]
This has to be fixed in 5.1.1 release https://rubygems.org/gems/haml/versions/5.1.1
Please let us know if you're still seeing this problem with 5.1.1 or newer.
